src/py/flwr/client/app.py

```diff
@@ -362,10 +366,13 @@ def _load_client_app() -> ClientApp:
                 if message is None:
                     time.sleep(3)  # Wait for 3s before asking again
                     continue
+                in_metadata = copy(message.metadata)
 
                 # Handle control message
                 out_message, sleep_duration = handle_control_message(message)
                 if out_message:
+                    if not validate_out_message(out_message, in_metadata):
+                        log(ERROR, "Invalid out message: %s", out_message)
                     send(out_message)
```

Comment:
Should we still send the message if the validation fails? Or rather construct and send a different error message?

Reply:
Oh, my bad. I thought `log(ERROR)` will interrupt the process. Yes, I think we should construct and send an error message instead of the invalid one.